better unicode handling

cubimon opened the pull request:
The user has to stop the unicode hex input with <kbd>return</kbd> or <kbd>space</kbd>, e.g. <kbd>compose key</kbd> <kbd>u</kbd> <kbd>2</kbd> <kbd>5</kbd> <kbd>6</kbd> <kbd>f</kbd> <kbd>space</kbd> result in "╯". This mimics the behaviour of ibus on linux, e.g. in chromium you can press ctrl+shift+u and enter a hex key (if you didn't modify default ibus key bindings). 

This solution removes the ugly prefix handling, but breaks compatibility (now you have to press enter to make the unicode character appear, which may require changes in other depending projects, e.g. qmk). 

This should fix #235

cubimon commented:
[this](https://github.com/cubimon/qmk_firmware/commit/106c24f9f2845d9acd7d06e3304fe066df0280b5) would fix qmk's unicode output

arkag commented:
Interested in getting this merged! Can we speed this along in anyway? @cubimon can you get the qmk changes merged, too?

cubimon commented:
@arkag I wrote @samhocevar an email last weekend about my pull request, because of responded to #257 (a newer pull request), but not mine. I haven't received an answert yet. So I just created a "work in progess" pull request on qmk, maybe someone over there is interested in this change.

samhocevar commented:
Hi! I am sorry for the delay but I am still unsure what to make of this. I would like WinCompose to act smoothly and allow for occasional breaking changes, whereas you want QMK to be robust and rely on a solid implementation.

What would you think about a completely different mechanism that would use a non-assigned key, such as `VK 0xE6` which is currently marked as [OEM Specific](https://docs.microsoft.com/en-us/windows/desktop/inputdev/virtual-key-codes)?
  - QMK would just send *e.g.* <kbd>VK.0xE6</kbd> <kbd>1</kbd> <kbd>2</kbd> <kbd>3</kbd> <kbd>Enter</kbd> for U+123
  - This would allow the user to have any key defined as the compose key, or even disable the compose key altogether

arkag commented:
As long as I get my (tm) and other various unicode strings working on windows with QMK, I don't really care how it's implemented!

cubimon commented:
Ok, I have no clue how I managed to get the submodules and I have deleted the whole repo after the push. Style checks apply to existing code, while sonar c# only checks added code? Anyway I added an additional unicode key, which behaves like the compose key, but accepts only hex values. So any unicode key can be input.

samhocevar commented:
Thanks for all your efforts and suggestions, @cubimon! I have finally decided that <kbd>Enter</kbd> would be mandatory to end manually entered Unicode sequences, as your initial patch suggested.

I still believe the QMK/WinCompose interface needs to be more robust and I will discuss the additional mechanism to go well with QMK somewhere else. While your new PR partially addresses the problem, it will not work on *e.g.* a Russian keyboard layout (see #261).
